### PR TITLE
Inversegaussian fit_mle

### PR DESCRIPTION
**Pull request by getzze**
Fit an InverseGaussian using sufficient statistics

### CODE REVIEW

File: src/univariate/continuous/inversegaussian.jl [hunk 1]
```diff
@@ -161,3 +161,37 @@ function rand(rng::AbstractRNG, d::InverseGaussian)
     u = rand(rng)
     u >= p1 ? μ^2 / x1 : x1
 end
+
+#### Fit model
+
+struct InverseGaussianStats <: SufficientStats
```
Comment (matbesancon) [inline src/univariate/continuous/inversegaussian.jl]: add documentation for `InverseGaussianStats`

File: src/univariate/continuous/inversegaussian.jl [hunk 2]
```diff
@@ -161,3 +161,37 @@ function rand(rng::AbstractRNG, d::InverseGaussian)
...
+    InverseGaussianStats(sx::Real, sinvx::Real, sw::Real) = new(sx, sinvx, sw)
+end
+
+function suffstats(::Type{InverseGaussian}, x::AbstractArray{T}) where T<:Real
+    sx = sum(x)
+    sinvx = sum(1 ./ x)
+    InverseGaussianStats(sx, sinvx, length(x))
+end
+
+function suffstats(::Type{InverseGaussian}, x::AbstractArray{T}, w::AbstractArray{Float64}) where T<:Real
```
Comment (matbesancon) [inline src/univariate/continuous/inversegaussian.jl]: you mean `AbstractArray` here? AbstractArray requires dimension

File: src/univariate/continuous/inversegaussian.jl [hunk 3]
```diff
@@ -161,3 +161,37 @@ function rand(rng::AbstractRNG, d::InverseGaussian)
...
+    sx = sum(w .* x)
+    sinvx = sum(w ./ x)
+    sw = sum(w)
+    InverseGaussianStats(sx, sinvx, sw)
+end
+
+function fit_mle(::Type{InverseGaussian}, ss::InverseGaussianStats)
+    mu = ss.sx / ss.sw
+    invlambda = ss.sinvx / ss.sw  -  1/mu
+    InverseGaussian(mu, 1 / invlambda )
```
Comment (matbesancon) [inline src/univariate/continuous/inversegaussian.jl]: use `inv(.)` instead of `1/.`

**Comment by getzze** [comment]
It is based on Gamma.jl and Normal.jl, I don't know enough about Julia to make changes to go beyond that.

**Comment by matbesancon** [comment]
@getzze check the individual comments I made you should be able to make modifications from them

File: src/univariate/continuous/inversegaussian.jl (hunk 1 above)
Comment (matbesancon) [reply src/univariate/continuous/inversegaussian.jl]: to add documentation, do:
"""
Sufficient statistics for the inference of an inverse Gaussian
"""
struct InverseGaussianStats <: SufficientStats

File: src/univariate/continuous/inversegaussian.jl (hunk 3 above)
Comment (matbesancon) [reply src/univariate/continuous/inversegaussian.jl]:
```suggestion
    invlambda = ss.sinvx / ss.sw - inv(mu)
    InverseGaussian(mu, inv(invlambda))
```

**Comment by matbesancon** [comment]
@getzze I made some specific suggestions on the changes to make

**Comment by matbesancon** [comment]
closing this in favor of 836

**Pull request closed without merging**